src/preproc/preprocessing_pass.h

```diff
@@ -0,0 +1,97 @@
+/*********************                                                        */
+/*! \file preprocessing_pass.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **  Justin Xu
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2017 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Implementation for preprocessing pass super class
+ **
+ ** Implementation for preprocessing pass super class. Includes a generalized
+ ** structure for the apply method, which includes dumping assertions before
+ ** and after the pass, initializing the Timer, and Tracing and Chatting. 
+ ** For new classes, a name is necessary to register the pass, and 
+ ** an apply internal method is necessary that takes in
+ ** the AssertionPipeline to perform the pass. An init Internal method
+ ** is optional to initialize variables that are not a part of the Context.
+ **/
+#include "cvc4_public.h"
+
+#ifndef __CVC4__PREPROC__PREPROCESSING_PASS_H
+#define __CVC4__PREPROC__PREPROCESSING_PASS_H
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "expr/node.h"
```

Comment:
Node is cvc4_private.h. Either switch this header to private or switch over to using Expr in these classes.

Reply:
After some thinking, I've switched the headers to be private now. It would probably be a bad idea to add the overhead of using `Expr`s for everything related to preprocessing. We could later introduce a special pass class that is public so that users can add their own passes.